schema/shared/src/main/scala/zio/blocks/schema/SchemaAspect.scala

```diff
@@ -0,0 +1,24 @@
+package zio.blocks.schema
+
+trait SchemaAspect[-Min, +Max] {
+
+  def apply[A](reflect: Reflect.Bound[A]): Reflect.Bound[A]
+}
+
+object SchemaAspect {
+
+  val identity: SchemaAspect[Any, Nothing] = new SchemaAspect[Any, Nothing] {
+    def apply[A](reflect: Reflect.Bound[A]): Reflect.Bound[A] = reflect
+  }
+
+  def doc(doc: String): SchemaAspect[Any, Nothing] = new SchemaAspect[Any, Nothing] {
+    def apply[A](reflect: Reflect.Bound[A]): Reflect.Bound[A] = reflect.doc(doc)
+  }
+
+  def examples[A](value: A, values: A*): SchemaAspect[A, A] = new SchemaAspect[A, A] {
+
+    def apply[A](reflect: Reflect.Bound[A]): Reflect.Bound[A] =
+      reflect.examples(value.asInstanceOf[A], values.asInstanceOf[Seq[A]]: _*)
```

Comment:
These casts should not be necessary.

Reply:
A is defined on 2 places here in `apply` of `SchemaAspect` and on `examples` method because it takes values of `A`. Compiler complains about 

```
type mismatch;
 found   : value.type (with underlying type A)
 required: A
 ```
 
if I move A to `trait SchemaAspect` level, than other aspects like identity would need a type parameter which would hurt user experience e.g. `@@ identity[Person]`

Reply:
When you add the type parameter bounds, you won't need the casts anymore.

Reply:
ok I renamed the first A to A0 and it works as you described